File: php-advisor-core/src/Payloads/GeometryPayload.php
<?php

namespace StormGeo\AdvisorCore\Payloads;

class GeometryPayload extends BasePayload
{
  /**
   * @var string
   */
	public $startDate;

  /**
   * @var string
   */
  public $endDate;

  /**
   * @var int
   */
  public $radius;

  /**
   * @var string
   */
  public $geometry;

  /**
   * @param array{startDate:string,endDate:string,radius:int,geometry:string} $parameters
   */
  public function __construct($parameters = [])
  {
    parent::__construct(
      ['startDate', 'endDate', 'radius', 'geometry'],
      $parameters
    );
  }

  /**
   * @var array
   */
  public function getQueryParams()
  {
    return [
      'startDate' => $this->startDate,
      'endDate' => $this->endDate,
      'radius' => $this->radius
    ];
  }

  /**
   * @var array
   */
  public function getBody()
  {
    return [
      'geometry' => $this->geometry
    ];
  }
}

File: dotnet-advisor-core/Payloads/GeometryPayload.cs
using System.Text;
using System.Text.Json;

namespace StormGeo.AdvisorCore.Payloads;

public class GeometryPayload
{
    public string? StartDate { get; set; }
    public string? EndDate { get; set; }
    public int? Radius { get; set; }
    public string? Geometry { get; set; }

    public string GetQueryParams()
    {
        return new QueryParameterBuilder()
            .AddStartDate(StartDate)
            .AddEndDate(EndDate)
            .AddRadius(Radius)
            .Build();
    }

    public HttpContent GetBody()
    {
        var json = JsonSerializer.Serialize(new Dictionary<string, string>() {
            { "geometry", Geometry ?? "" },
        });

        return new StringContent(json, Encoding.UTF8, "application/json");
    }
}
